### PR TITLE
Explaination of leftout conformity checks in submitSolution

### PR DESCRIPTION
**Pull request by josojo**
Alan requested these comments for better understanding

### CODE REVIEW

File: .solhint.json [hunk 1]
```diff
@@ -4,7 +4,7 @@
     "indent": ["error", 4],
     "quotes": ["error", "double"],
     "max-line-length": ["error", 129],
-    "function-max-lines": ["error", 61],
+    "function-max-lines": ["error", 65],
```
Comment (fleupold) [inline .solhint.json]: I really dislike that we keep pushing this value in each PR. That kind of makes this rule pointless. Can we just remove the max lines constraint?

File: contracts/BatchExchange.sol [hunk 2]
```diff
@@ -293,6 +293,10 @@ contract BatchExchange is EpochTokenLocker {
         require(tokenIdsForPrice[0] != 0, "Fee token has fixed price!");
         require(tokenIdsForPrice.checkPriceOrdering(), "prices are not ordered by tokenId");
         require(owners.length <= MAX_TOUCHED_ORDERS, "Solution exceeds MAX_TOUCHED_ORDERS");
+        // Further assumptions are: owners.length == orderIds.length && owners.length == buyVolumes.length
+        // && prices.length == tokenIdsForPrice.length
+        // These assumptions are not checked explicitly, as violations of these constraints can not be used
```
Comment (fleupold) [inline contracts/BatchExchange.sol:298]: Do we have tests trying to exploit an out of bounds situation?

**Review by fleupold — APPROVED** [review]
Many arrays which are implicitly aligned 

<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia1.giphy.com%2Fmedia%2Fl2Je8hWa25y5dUKsg%2Fgiphy.gif&token=complicated" data-gitmeme-token="complicated"></a>

File: contracts/BatchExchange.sol (hunk 2 above)
Comment (bh2smith) [reply contracts/BatchExchange.sol:298]: Good question! I think we should. I may have included one or two of these at some point for other functions, but possibly not for `submitSolution`.

Ahhh yes... `placeOrders` has tests with array index out of bound errors.

File: .solhint.json (hunk 1 above)
Comment (bh2smith) [reply .solhint.json]: YES!

**Review by bh2smith — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia3.giphy.com%2Fmedia%2F6pSJGZW0ZmbUA%2Fgiphy.gif&token=wicked" data-gitmeme-token="wicked"></a>

**Review by bh2smith — APPROVED** [review]
This is fine for me if it works. However I would expect that simply removing this `function-max-lines` configuration will just fail because the internal default is much less than what we have been allowing.

**Comment by bh2smith** [comment]
How can this be merged if the last commit was failing?

**Pull request merged**